public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -227,6 +230,89 @@ angular.module('jasmic.controllers')
                 $scope.demand.de_until = moment().add(7, 'days').toDate();
             };
 
+            $scope.editDemandForm = function (obj) {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                $scope.demand = obj;
+
+                //for some reason the dates needed to be passed through MomentJS as seen below to be assigned to the form inputs
+
+                $scope.demand.de_posting_date= moment(obj.de_posting_date).toDate();
+
+                $scope.demand.de_until= moment(obj.de_until).toDate();
+
+            }
+
+            $scope.updateDemand = function() {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                DemandEditFactory.update({id:$scope.demands[0].bu_buyer._id, demand_id:$scope.demand._id}, $scope.demand, function(success) {
```

Comment:
@nick22891 any hardcoding of array elements is discouraged.  Never do this:
```javascript
$scope.demands[0].bu_buyer._id
```
There are other ways of accessing the `bu_buyer._id`.